### PR TITLE
aaf/1.2.0: Add interface definition for Windows

### PR DESCRIPTION
**Pull request by MartinDelille**
Specify library name and version:  **aaf/1.2.0**

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by garethsb** [comment]
Why is this required in the conan package but not in upstream?

**Comment by MartinDelille** [comment]
I submit the fix (and some of the other patches) to the upstream repo: https://sourceforge.net/p/aaf/code2/merge-requests/1/

**Review by SSE4 — DISMISSED** [review]

**Review by uilianries — DISMISSED** [review]

File: recipes/aaf/all/patches/0008-define-interface-for-windows.patch [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+diff --git a/ref-impl/include/ref-api/AAF.h b/ref-impl/include/ref-api/AAF.h
+index 058e325cc..6d93d0f97 100644
+--- a/ref-impl/include/ref-api/AAF.h
++++ b/ref-impl/include/ref-api/AAF.h
+@@ -49,6 +49,11 @@
+ #endif
+ 
+ #ifdef __cplusplus
++
++#if defined (_WINDOWS_) && !defined (interface)
```
Comment (SpaceIm) [inline recipes/aaf/all/patches/0008-define-interface-for-windows.patch:16]: What is `_WINDOWS_`? Is it defined in some headers of `aaf`?

Comment (garethsb) [reply recipes/aaf/all/patches/0008-define-interface-for-windows.patch:16]: Ha! I think AAF SDK defines `OS_WINDOWS` in fact (based on `_WIN32`).

https://sourceforge.net/p/aaf/code2/ci/master/tree/ref-impl/include/AAFPlatform.h

Comment (SpaceIm) [reply recipes/aaf/all/patches/0008-define-interface-for-windows.patch:16]: yes, `_WINDOWS_` doesn't exist in source code.

Moreover, `interface` is defined in `AAFCOMPlatformTypes.h`, so if there is some issue when `OS_WINDOWS` is defined, it should be fixed in this file.

**Comment by SpaceIm** [comment]
Side comment: why version is 1.2.0 and not 1.2.0-DR1? I don't see any comment about this inconsistency in original PR.

**Comment by SSE4** [comment]
> Side comment: why version is 1.2.0 and not 1.2.0-DR1? I don't see any comment about this inconsistency in original PR.

@SpaceIm what is DR1?

**Comment by SpaceIm** [comment]
> > Side comment: why version is 1.2.0 and not 1.2.0-DR1? I don't see any comment about this inconsistency in original PR.
> 
> @SpaceIm what is DR1?

No idea, but it's like a beta version i guess, before Release Candidate: see history of archives https://sourceforge.net/projects/aaf/files/AAF-src/

File: recipes/aaf/all/patches/0008-define-interface-for-windows.patch (hunk 1 above)
Comment (SSE4) [reply recipes/aaf/all/patches/0008-define-interface-for-windows.patch:16]: `_WINDOWS_` is defined in `Windows.h` (part of Windows SDK):
it's basically an include guard for that file.
if you're looking at pre-defined macro by the compiler (which is available regardless of Windows SDK headers inclusion), you're probably looking for `_WIN32`.

**Comment by SSE4** [comment]
> No idea, but it's like a beta version i guess, before Release Candidate: see history of archives https://sourceforge.net/projects/aaf/files/AAF-src/

if it's the case, we don't need to package pre-release (DR1). just release 1.2.0 is enough, especially if it's already available.

**Comment by garethsb** [comment]
> if it's the case, we don't need to package pre-release (DR1). just release 1.2.0 is enough, especially if it's already available.

It's not.

File: recipes/aaf/all/patches/0008-define-interface-for-windows.patch (hunk 1 above)
Comment (garethsb) [reply recipes/aaf/all/patches/0008-define-interface-for-windows.patch:16]: `interface` is only defined in [AAFCOMPlatformTypes.h](https://sourceforge.net/p/aaf/code2/ci/master/tree/ref-impl/include/AAFCOMPlatformTypes.h#l121) if `OS_UNIX` (and not `AAF_OMIT_WINDOWS_DATA_TYPES_DEFINITIONS`). On Windows, the `#define interface struct` should come from Windows SDK headers, I thought?

Comment (MartinDelille) [reply recipes/aaf/all/patches/0008-define-interface-for-windows.patch]: I apply the patch to `AAFCOMPlatformTypes.h`.

**Review by SSE4 — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Comment by garethsb** [comment]
@MartinDelille, I still don't understand why this patch is required.

_AAFCOMPlatformTypes.h_ has `#include <wtypes.h>` on Windows which on my system leads to `#define interface __STRUCT__` (which is `struct`) via:

Line 21 in _C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\shared\wtypes.h_
Line 20 in _C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\shared\rpc.h_
Line 217 in _C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\um\windows.h_
Line 37 in _C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\um\ole2.h_
Line 27 in _C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\um\objbase.h_
Line 169 in _C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\um\combaseapi.h_

**Comment by MartinDelille** [comment]
@garethsb Indeed: My project using AAF was undefining interface elsewhere.

**Pull request closed without merging**